lib/puppet/pops/types/puppet_object.rb

```diff
@@ -0,0 +1,21 @@
+module Puppet::Pops
+module Types
+
+# Marker module for implementations that are mapped to Object types
+# @api public
+module PuppetObject
+  # Returns all classes that includes this module
+  def self.descendants
+    ObjectSpace.each_object(Class).select { |klass| klass < self }
```

Comment:
Where/how is this used? Looks like it could potentially be very slow (iterates over everything in memory)